### PR TITLE
Move methods from config companion object to object Config

### LINKED ISSUE #28: Move the methods from config package to the companion object Config
Move: `int`, `double`, `string`,`long`, `short`,`uri`, `read`, `write`, and `report` to the companion object `Config` and make sure the examples still work.

### PR DESCRIPTION
**Pull request by pgabara**
Fixes #28

### CODE REVIEW

File: zio-config/src/main/scala/zio/config/Config.scala [hunk 1]
```diff
@@ -37,4 +41,21 @@ object Config {
...
+
+  def short(path: String): Config[Short] = Config.Source(path, PropertyType.ShortType)
+
+  def uri(path: String): Config[URI] = Config.Source(path, PropertyType.UriType)
+
+  def read[A](config: => Config[A]): Read[A] = Read.read[A](config)
+
+  def write[A](config: => Config[A]): Write[A] = Write.write[A](config)
+
+  def report[A](config: => Config[A]): Report[A] = Report.report[A](config)
```
Comment (afsalthaj) [inline zio-config/src/main/scala/zio/config/Config.scala]: I would imagine the `read`, `write` and `report` to be still part of the package object and rest of the functions here.

The main import for the user could be , for example,
`import zio.config._, Config._`

File: zio-config/src/main/scala/zio/config/package.scala [hunk 2]
```diff
@@ -20,10 +11,6 @@ package object config extends Sources {
       })
       .onError(_ => None)
 
-  def read[A](config: => Config[A]): Read[A]     = Read.read[A](config)
-  def write[A](config: => Config[A]): Write[A]   = Write.write[A](config)
-  def report[A](config: => Config[A]): Report[A] = Report.report[A](config)
```
Comment (afsalthaj) [inline zio-config/src/main/scala/zio/config/package.scala]: Based on the discussions, the read write and report are back in here. We can inline the functions later on though.
@wi101?

File: zio-config/src/main/scala/zio/config/Config.scala (hunk 1 above)
Comment (pgabara) [reply zio-config/src/main/scala/zio/config/Config.scala]: sure, `read`, `write` and `report` moved back.

File: zio-config/src/main/scala/zio/config/package.scala (hunk 2 above)
Comment (wi101) [reply zio-config/src/main/scala/zio/config/package.scala]: Alright

**Comment by afsalthaj** [comment]
@pgabara If you could resolve the conflicts, we can merge this one ?

**Comment by pgabara** [comment]
I resolved conflicts and added new functions: `byte`, `bigInt`, `boolean` etc to companion object.

**Pull request merged**